### PR TITLE
Move utility code into `utils` directory

### PR DESCRIPTION
**Pull request by avi-perl**
This PR moves general code into a `utils` directory which:
- Cleans up the project root.
- Removes code from the apps folder that, strictly speaking, are not apps, but is code that apps are dependent on. With this change, files in the apps folder will truly only contain "apps".
- Puts dependency code into an expected location.

### CODE REVIEW

**Review by kbaskett248 — CHANGES_REQUESTED** [review]
Hey @avi-perl I really like the way this is going. It seems that you didn't remove the old versions of the files from their previous locations, though.

Also, would you mind running `black` and `isort` on the files. I'll work on adding that in with `precommit` so we don't have to think about it.

File: apps/base.py [hunk 1]
```diff
@@ -13,8 +13,7 @@
 import terminalio
 from adafruit_display_shapes.rect import Rect
 from adafruit_display_text import label
-
-from app_pad import AppPad, DoubleTapEvent, EncoderEvent, EncoderButtonEvent, KeyEvent
+from app_pad import AppPad, DoubleTapEvent, EncoderButtonEvent, EncoderEvent, KeyEvent
```
Comment (kbaskett248) [inline apps/base.py]: This file should also be deleted, correct?

File: apps/key.py [hunk 2]
```diff
@@ -5,17 +5,15 @@
 """
 
 try:
-    from typing import Any, Dict, List, Set, Optional, Union
+    from typing import Any, Dict, List, Optional, Set, Union
```
Comment (kbaskett248) [inline apps/key.py:8]: Shouldn't this file also be deleted?

**Review by kbaskett248 — CHANGES_REQUESTED** [review]
👍  Cool, almost there. Found two more files that need deleting. I went ahead and tested with those files deleted and everything works great!

File: apps/base.py (hunk 1 above)
Comment (avi-perl) [reply apps/base.py]: You are quite right, missed it!
Removed.

File: apps/key.py (hunk 2 above)
Comment (avi-perl) [reply apps/key.py:8]: You are quite right, missed it!
Removed.

**Review by kbaskett248 — APPROVED** [review]
💯 Awesome! Thanks!

**Pull request merged**